### PR TITLE
Upgrade Playwright to v1.45

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Upgrade Playwright to v1.45

## What's new?
https://github.com/microsoft/playwright/releases/tag/v1.44.0
https://github.com/microsoft/playwright/releases/tag/v1.45.0

## Testing Instructions
* Run any e2e test locally; the command should download new browsers and run the test successfully.
* CI checks are green.

### CODE REVIEW

File: test/e2e/specs/site-editor/navigation-editor.spec.js [hunk 1]
```diff
@@ -62,11 +62,9 @@ test.describe( 'Editing Navigation Menus', () => {
...
 
 			// The Navigation block description should contain the locked state information.
-			const navBlockNodeDescriptionId =
-				await navBlockNode.getAttribute( 'aria-describedby' );
-			await expect(
-				listView.locator( `id=${ navBlockNodeDescriptionId }` )
-			).toHaveText( /This block is locked./ );
+			await expect( navBlockNode ).toHaveAccessibleDescription(
+				/This block is locked./
+			);
```
Comment (Mamaduka) [inline test/e2e/specs/site-editor/navigation-editor.spec.js:67]: @afercia, I think you'll like the new a11y assertions - https://github.com/microsoft/playwright/releases/tag/v1.44.0

**Review by swissspidy — APPROVED** [review]

File: test/e2e/specs/site-editor/navigation-editor.spec.js (hunk 1 above)
Comment (afercia) [reply test/e2e/specs/site-editor/navigation-editor.spec.js:67]: Uhu, thanks for pointing me at it 🥳

**Comment by Mamaduka** [comment]
The RichText e2e tests seem to be failing on Firefox after the upgrade. I couldn't reproduce them manually, but I will continue debugging tomorrow.

**Comment by Mamaduka** [comment]
Some of the `RichText` e2e tests are failing, and I wasn't able to track down the exact issue. It looks like the focus (Firefox) and some keys (Webkit) behave differently, which could be related to the browser updates that shipped with this version.

**Comment by WunderBart** [comment]
> Some of the `RichText` e2e tests are failing, and I wasn't able to track down the exact issue. It looks like the focus (Firefox) and some keys (Webkit) behave differently, which could be related to the browser updates that shipped with this version.

@Mamaduka, I'm wondering if we should use [`pressSequentially`](https://playwright.dev/docs/api/class-locator#locator-press-sequentially) instead of discouraged `type` in the RichText tests. For one, it seems to first focus the element which doesn't seem to be the case for `type`. If that doesn't help, maybe we could try adding a little bit of delay to simulate actual user typing / let the UI catch up?

**Comment by Mamaduka** [comment]
@WunderBart, do you think there are changes in `page.type` with the latest PW release that are causing the failures? It is either that or browser updates. Hopefully, v1.45 will be released soon, and I can test the theory.

**Comment by WunderBart** [comment]
> @WunderBart, do you think changes in `page.type` with the latest Playwright release are causing the failures? It seems either that or browser updates could be the issue.

I'm not sure, but considering that `page.type` requires the element to be focused before typing starts, combined with the RichText tests _assuming_ the focus is always correctly set, I think switching to `locator.pressSequentially` might help, regardless of whether the issue is with the browser or Playwright. 😄 🤞

**Comment by Mamaduka** [comment]
The same failures remain after the latest update; there might be an actual bug with Firefox and RichText. I'll try to debug it later.

* should return focus when pressing formatting button
* should update internal selection after fresh focus

@WunderBart, I don't think we can use `locator.pressSequentially` in this case because the focus needs to be handled by the app and not the test runner.

**Comment by Mamaduka** [comment]
@ellatrix, I'm thinking to skip [this two failing](https://github.com/WordPress/gutenberg/pull/61443#issuecomment-2191347625) RichText e2e tests on Firefox.

* They only fail when running the project with Firefox. This could be related to the updates that Playwright is shipping.
* Unable to reproduce it manually and not debug because I'm limited to Playwright's UI Mode.
* Both tests are related to the focus returning to the RichText value.

**Comment by ellatrix** [comment]
Sure let's skip Firefox for these two

**Pull request merged**